server/src/main/java/io/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -169,9 +170,20 @@ public JacksonSmileProvider getJacksonSmileProvider(@Smile ObjectMapper objectMa
 
   static Server makeJettyServer(DruidNode node, ServerConfig config, TLSServerConfig tlsServerConfig)
   {
-    final QueuedThreadPool threadPool = new QueuedThreadPool();
-    threadPool.setMinThreads(config.getNumThreads());
-    threadPool.setMaxThreads(config.getNumThreads());
+    final QueuedThreadPool threadPool;
+    if (config.getQueueSize() == Integer.MAX_VALUE) {
```

Comment:
i think its fine given that its a static constant already and just really checking whether queue is unbounded or bounded.